lib/govuk_app_config/govuk_error/govuk_data_sync.rb

```diff
@@ -0,0 +1,56 @@
+require "time"
+
+module GovukError
+  class GovukDataSync
+    class MalformedDataSyncPeriod < StandardError
+      attr_reader :invalid_value
+
+      def initialize(invalid_value)
+        @invalid_value = invalid_value
+      end
+
+      def message
+        "\"#{invalid_value}\" is not a valid value (should be of form '22:00-03:00')."
+      end
+    end
+
+    class MissingDataSyncPeriod < StandardError
+      def message
+        "You must specify a ENV['GOVUK_DATA_SYNC_PERIOD'] (should be of form '22:00-03:00')."
+      end
+    end
+
+    attr_reader :from, :to
+
+    def initialize(govuk_data_sync_period)
+      raise MissingDataSyncPeriod if govuk_data_sync_period.nil?
```

Comment:
I don't think we should raise an error on this and should use whether the env var is defined or not to determine whether we're doing this.

My logic is that all our apps will have the same code running in prod, staging and integration and the differentiator between prod and the others will be the presence of this env var.